### PR TITLE
Emit InvalidArgument on CopyPart invalid range

### PR DESCRIPTION
**Pull request by andrewgaul**
References kahing/goofys#212.

### CODE REVIEW

**Comment by andrewgaul** [comment]
@kahing Please review.

File: src/main/java/org/gaul/s3proxy/S3ProxyHandler.java [hunk 1]
```diff
@@ -2221,6 +2224,10 @@ private void handleCopyPart(HttpServletRequest request,
         }
 
         BlobMetadata blobMetadata = blob.getMetadata();
+        // HTTP GET allow overlong ranges but S3 CopyPart does not
+        if (expectedSize != -1 && blobMetadata.getSize() < expectedSize) {
```
Comment (kahing) [inline src/main/java/org/gaul/s3proxy/S3ProxyHandler.java]: what happens if I request 5-6 for a blob that has 4 bytes?

Comment (andrewgaul) [inline src/main/java/org/gaul/s3proxy/S3ProxyHandler.java]: Normal GET handling will reject the invalid beginning.  This code only needs to add the additional CopyPart checks.

Comment (kahing) [inline src/main/java/org/gaul/s3proxy/S3ProxyHandler.java]: what about 4-6 for a blob that's 5 bytes?

Comment (andrewgaul) [inline src/main/java/org/gaul/s3proxy/S3ProxyHandler.java]: The size check will catch that.

Comment (kahing) [inline src/main/java/org/gaul/s3proxy/S3ProxyHandler.java]: how so? expectedSize = 6-4+1 = 3, blobMetadata.getSize() will be 5.

Comment (andrewgaul) [inline src/main/java/org/gaul/s3proxy/S3ProxyHandler.java]: S3Proxy passes the range to `getBlob` so that getSize will return 1 here.

**Pull request merged**